### PR TITLE
General cleanup, remove `root_finding` and `kalman` modules

### PR DESCRIPTION
**Pull request by dwillmer**
New Feature

- [X] Code conforms to style guide
- [ ] Doctests
- [X] Unit-tests
- [ ] Documentation
- [ ] New line item in [releases.md](https://github.com/fastats/fastats/blob/master/releases.md)
- [ ] Info added to `literature` directory

### CODE REVIEW

**Comment by dwillmer** [comment]
@fastats/core - ready for review

File: tests/optimize/test_binary_search.py [hunk 1]
```diff
@@ -5,8 +5,8 @@
 from numpy import cos, tan
 from pytest import approx
 
-from fastats.optimise.root_finding import binary_search
-from fastats.optimise.root_finding.binary_search import root
+from fastats.optimise.binary_search import binary_search
```
Comment (rjenc29) [inline tests/optimize/test_binary_search.py]: Could possibly `import binary_search, root`

File: tests/optimize/test_newton_raphson.py [hunk 2]
```diff
@@ -4,8 +4,8 @@
 from numpy import cos, tan
 from pytest import approx, raises
 
-from fastats.optimise.root_finding import newton_raphson
-from fastats.optimise.root_finding.newton_raphson import root
+from fastats.optimise.newton_raphson import newton_raphson
```
Comment (rjenc29) [inline tests/optimize/test_newton_raphson.py]: Could possibly import `newton_raphson, root`

File: tests/optimize/test_binary_search.py (hunk 1 above)
Comment (dwillmer) [reply tests/optimize/test_binary_search.py]: Indeed, nice catch, thanks!

PR updated.

File: tests/optimize/test_newton_raphson.py (hunk 2 above)
Comment (dwillmer) [reply tests/optimize/test_newton_raphson.py]: Ditto, PR updated, ta muchly!

**Pull request merged**